src/core/dbus-execute.c

```diff
@@ -2808,6 +2843,56 @@ int bus_exec_context_set_transient_property(
                         return 1;
                 }
 
+        } else if (streq(name, "MountImages")) {
+                char *source, *destination;
+                int permissive;
+                bool empty = true;
+
+                r = sd_bus_message_enter_container(message, 'a', "(ssb)");
+                if (r < 0)
+                        return r;
+
+                while ((r = sd_bus_message_read(message, "(ssb)", &source, &destination, &permissive)) > 0) {
+                        if (!path_is_absolute(source))
+                                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Source path %s is not absolute.", source);
+                        if (!path_is_absolute(destination))
+                                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Destination path %s is not absolute.", destination);
+
+                        if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
+                                r = mount_images_add(&c->mount_images, &c->n_mount_images,
+                                                   &(MountEntry) {
+                                                           .source_const = source,
+                                                           .path_const = destination,
+                                                           .ignore = permissive,
+                                                   });
+                                if (r < 0)
+                                        return r;
+
+                                unit_write_settingf(
+                                                u, flags|UNIT_ESCAPE_SPECIFIERS, name,
+                                                "%s=%s%s:%s",
+                                                name,
+                                                permissive ? "-" : "",
+                                                source,
+                                                destination);
```

Comment:
this will fall apart if "source" or "destination" contains a ":". maybe use C escaping when writing this out here and using EXTRACT_CUNESCAPE when parsing.

Reply:
Good point, fixed and added new unit test cases to cover this

Reply:
Existing options will suffer from the same issue, like BindPaths/BindReadOnlyPaths